src/App.tsx

```diff
@@ -1,61 +1,109 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
 
 export const App = () => {
+  const [userSelect, setUserSelect] = useState('0');
+  const [title, setTitile] = useState('');
+  const [todos, setTodos] = useState(todosFromServer);
+  const [submit, setSubmit] = useState(true);
+
+  const trimedTitle = title.trim();
+
+  const largestTodoId = todos.reduce((previously, currently) => (
+    previously.id > currently.id ? previously : currently
+  ));
+
+  const updateTodoList = () => {
+    const newTodo = {
+      id: largestTodoId.id + 1,
+      title: trimedTitle,
+      completed: false,
+      userId: Number(userSelect),
+    };
+
+    setTodos(old => (
+      [...old, newTodo]
+    ));
+    setTitile('');
+    setUserSelect('0');
+    setSubmit(true);
+  };
+
+  const handleSubmit = (event:React.MouseEvent) => {
+    setSubmit(false);
+    event.preventDefault();
+    if (trimedTitle !== '' && userSelect !== '0') {
+      setSubmit(true);
+
+      return updateTodoList();
+    }
+
+    setSubmit(false);
+
+    return null;
```

Comment:
1. prevent event
2. check if condition false - set submit to false and return from function
3. if your condition is true just call update list (set submit is already there)

You don't need repeated code and returns.